x/ibc/07-tendermint/types/misbehaviour_handle.go

```diff
@@ -25,29 +25,29 @@ func (cs ClientState) CheckMisbehaviourAndUpdateState(
 	clientStore sdk.KVStore,
 	misbehaviour clientexported.Misbehaviour,
 ) (clientexported.ClientState, error) {
+	tmEvidence, ok := misbehaviour.(*Misbehaviour)
+	if !ok {
+		return nil, sdkerrors.Wrapf(clienttypes.ErrInvalidClientType, "expected type %T, got %T", misbehaviour, &Misbehaviour{})
+	}
 
 	// If client is already frozen at earlier height than misbehaviour, return with error
-	if cs.IsFrozen() && cs.FrozenHeight <= uint64(misbehaviour.GetHeight()) {
+	height := clienttypes.NewHeight(0, uint64(misbehaviour.GetHeight()))
+	if cs.IsFrozen() && !cs.FrozenHeight.GT(height) {
```

Comment:
this is kinda hard to read. This leans me to think we should just add `LTE` and `GTE` for readability since this needs to be secure and can be easy to use the wrong logic here